types/type_map.go

```diff
@@ -99,3 +103,16 @@ func (t typeMap) Validate() error {
 	}
 	return t.value.Validate()
 }
+
+func genName(baseName string, key bool) string {
+	postfix := "value"
+	if key {
+		postfix = "key"
+	}
+	digest := md5.New()
+	n, err := digest.Write([]byte(baseName + postfix))
+	if err != nil {
+		log.Fatalf("failed to build map key/value name for field %s", baseName)
+	}
+	return fmt.Sprintf("%s%x", postfix, n)
```

Comment:
Actually I don't get what here is happened. Do you count bytes with `md5` package?